diff --git a/src/raylib.h b/src/raylib.h
index 8a5d50c2..bd3446db 100644
--- a/src/raylib.h
+++ b/src/raylib.h
@@ -207,10 +207,10 @@ typedef struct Vector4 {
     float y;                // Vector y component
     float z;                // Vector z component
     float w;                // Vector w component
-} Vector4;
+} Quaternion;
 
 // Quaternion, 4 components (Vector4 alias)
-typedef Vector4 Quaternion;
+typedef Quaternion Vector4;
 
 // Matrix, 4x4 components, column major, OpenGL style, right handed
 typedef struct Matrix {
@@ -252,10 +252,10 @@ typedef struct Texture {
     int height;             // Texture base height
     int mipmaps;            // Mipmap levels, 1 by default
     int format;             // Data format (PixelFormat type)
-} Texture;
+} Texture2D;
 
 // Texture2D, same as Texture
-typedef Texture Texture2D;
+typedef Texture2D Texture;
 
 // TextureCubemap, same as Texture
 typedef Texture TextureCubemap;
@@ -265,10 +265,10 @@ typedef struct RenderTexture {
     unsigned int id;        // OpenGL framebuffer object id
     Texture texture;        // Color buffer attachment texture
     Texture depth;          // Depth buffer attachment texture
-} RenderTexture;
+} RenderTexture2D;
 
 // RenderTexture2D, same as RenderTexture
-typedef RenderTexture RenderTexture2D;
+typedef RenderTexture2D RenderTexture;
 
 // NPatchInfo, n-patch layout info
 typedef struct NPatchInfo {
@@ -400,6 +400,8 @@ typedef struct ModelAnimation {
     Transform **framePoses; // Poses array by frame
 } ModelAnimation;
 
+typedef Transform* TransformPtr;
+
 // Ray, ray for raycasting
 typedef struct Ray {
     Vector3 position;       // Ray position (origin)
diff --git a/src/raymath.h b/src/raymath.h
index fbe4cead..8e1012fd 100644
--- a/src/raymath.h
+++ b/src/raymath.h
@@ -47,6 +47,8 @@
 #ifndef RAYMATH_H
 #define RAYMATH_H
 
+#define RAYMATH_IMPLEMENTATION
+
 #if defined(RAYMATH_IMPLEMENTATION) && defined(RAYMATH_STATIC_INLINE)
     #error "Specifying both RAYMATH_IMPLEMENTATION and RAYMATH_STATIC_INLINE is contradictory"
 #endif
@@ -54,11 +56,11 @@
 // Function specifiers definition
 #if defined(RAYMATH_IMPLEMENTATION)
     #if defined(_WIN32) && defined(BUILD_LIBTYPE_SHARED)
-        #define RMAPI __declspec(dllexport) extern inline // We are building raylib as a Win32 shared library (.dll).
+        #define RMAPI __declspec(dllexport) extern // We are building raylib as a Win32 shared library (.dll).
     #elif defined(_WIN32) && defined(USE_LIBTYPE_SHARED)
         #define RMAPI __declspec(dllimport)         // We are using raylib as a Win32 shared library (.dll)
     #else
-        #define RMAPI extern inline // Provide external definition
+        #define RMAPI extern // Provide external definition
     #endif
 #elif defined(RAYMATH_STATIC_INLINE)
     #define RMAPI static inline // Functions may be inlined, no external out-of-line definition
diff --git a/src/rcore.c b/src/rcore.c
index 7acedf6c..4c8676aa 100644
--- a/src/rcore.c
+++ b/src/rcore.c
@@ -110,6 +110,10 @@
 
 #include "raylib.h"                 // Declares module functions
 
+#if !defined(GLFW_MOUSE_PASSTHROUGH)
+    #define GLFW_MOUSE_PASSTHROUGH      0x0002000D
+#endif
+
 // Check if config flags have been externally provided on compilation line
 #if !defined(EXTERNAL_CONFIG_FLAGS)
     #include "config.h"             // Defines module configuration flags
@@ -120,8 +124,10 @@
 #define RLGL_IMPLEMENTATION
 #include "rlgl.h"                   // OpenGL abstraction layer to OpenGL 1.1, 3.3+ or ES2
 
-#define RAYMATH_IMPLEMENTATION      // Define external out-of-line implementation
-#include "raymath.h"                // Vector3, Quaternion and Matrix functionality
+#ifndef RAYMATH_IMPLEMENTATION
+    #define RAYMATH_IMPLEMENTATION      // Define external out-of-line implementation
+    #include "raymath.h"                // Vector3, Quaternion and Matrix functionality
+#endif
 
 #if defined(SUPPORT_GESTURES_SYSTEM)
     #define GESTURES_IMPLEMENTATION
